src/main.c

```diff
@@ -280,6 +282,12 @@ static void workaround_sync_event_detail_with_week_view(GtkWidget* event_panel,
 	gtk_widget_set_size_request(event_panel, allocation->width, allocation->height);
 }
 
+static gboolean sync_func(gpointer userdata)
+{
+	do_calendar_sync((FocalApp*) userdata);
```

Comment:
GObject-based objects should be cast with their helper macros, i.e. `FOCAL_APP(userdata)`. This allows a runtime check that the object is indeed a `FocalApp`. There are plenty of examples of this throughout this file

Reply:
Noted, and neat! Thanks, will update accordingly!